Added changelog (#157)
timescale/docs: timescaledb/overview/release-notes/index.md

For our next release, we plan to add:

- Elasticity: Move data when expanding or shrinking a multi-node cluster

> - Experimental Schema: A way to get experimental functionalities early to our users

Reviewer: ```suggestion
- Experimental schema: A way to get experimental functionalities early to our users
```

- Various bug fixes.

> The current GA (Generally Available) version is 2.3.1.

Reviewer: ```suggestion
The current generally available (GA) version is 2.3.1.
```

behaviors to suit your needs.  
Some of these API updates are **breaking changes**.

### What's new in TimescaleDB 2.3.1:

This maintenance release adds support for downgrading Timescale DB 2.3.1 to 2.3.

> We intend to keep expanding this functionality as we release new versions to allow users to go down 1 version at a time.

Reviewer: ```suggestion
We intend to keep expanding this functionality as we release new versions to allow users to downgrade one version at a time.
```

> Several bugfixes addressed, please see the release notes below for more details.

Reviewer: ```suggestion
Several bugs fixed, see the release notes for more details.
```

**PostgreSQL 11 deprecation announcement**

Timescale is working hard on our next exciting features. 
To make that possible, we require functionality 
that is available in Postgres 12 and above. 

> For this reason, TimescaleDB 2.3.1 will be the final version that supports PostgreSQL 11.  

Reviewer: ```suggestion
For this reason, TimescaleDB 2.3.1 is the final version that supports PostgreSQL 11.  
```

## Unreleased

 ## 2.3.1 (2021-07-05)

> This maintenance release contains bugfixes since the 2.3.0 release. 

Reviewer: ```suggestion
This maintenance release contains bug fixes since the 2.3.0 release. 
```

We deem it moderate priority for upgrading.

> **Bugfixes**

Reviewer: ```suggestion
**Bug fixes**
```
